### ISSUE TITLE
Aborted (core dumped) with tf.raw_ops.AvgPoolGrad

### ISSUE THREAD

**Issue by vwrewsge** [tensorflow/tensorflow#63353]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

binary

### TensorFlow version

tf 2.15

### Custom code

Yes

### OS platform and distribution

Ubuntu 20.04

### Python version

3.9

### Current behavior?

core dumped error with specific input parameters.

### Standalone code to reproduce the issue

```shell
import tensorflow as tf

# Generate input data
input_data = tf.random.normal([1, 28, 28, 3])
grad = tf.random.normal([1, 14, 14, 6])  # Change the number of channels in grad tensor

# Perform average pooling
result = tf.nn.avg_pool2d(input_data, ksize=[1, 2, 2, 1], strides=[1, 2, 2, 1], padding='VALID', data_format='NHWC')

# Compute gradient
grad_result = tf.raw_ops.AvgPoolGrad(orig_input_shape=tf.shape(input_data), grad=grad, ksize=[1, 2, 2, 1], strides=[1, 2, 2, 1], padding='VALID', data_format='NHWC')

print(grad_result)
```

### Relevant log output

```shell
free(): corrupted unsorted chunks
Aborted (core dumped)
```

**Comment by sushreebarsa** [comment]
@sachinprasadhs I was able to replicate the error reported [here](https://colab.research.google.com/gist/sushreebarsa/fa1b2962c80219519380bfebc99f9879/63353.ipynb). Thank you!

**Comment by gzmkl** [comment]
Not a problem any more. I verified with the current TF 2.21 master branch

**Comment by Kayyuri** [comment]
Hi @vwrewsge, 
Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The crash occurs due to shape mismatch. ` tf.raw_ops.AvgPoolGrad` expects` grad.shape` to exactly match the pooling output shape.  In the provided code AvgPool2D produces a 3-channel output, but your grad tensor provides  6 channels, results in crash. To avoid this, update grad to `[1, 14, 14, 3]` , so the gradient shape matches the pooling output.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
